lib/ansible/modules/cloud/amazon/ec2_vol_facts.py

```diff
@@ -59,57 +60,65 @@
 import traceback
 
 try:
-    import boto.ec2
-    from boto.exception import BotoServerError
-    HAS_BOTO = True
+    from botocore.exceptions import ClientError
+    from botocore.exceptions import ProfileNotFound
 except ImportError:
-    HAS_BOTO = False
+    pass  # caught by imported HAS_BOTO3
 
 from ansible.module_utils.basic import AnsibleModule
-from ansible.module_utils.ec2 import connect_to_aws, ec2_argument_spec, get_aws_connection_info
+from ansible.module_utils.ec2 import connect_to_aws, ec2_argument_spec, get_aws_connection_info, boto3_conn, HAS_BOTO3, boto3_tag_list_to_ansible_dict
```

Comment:
You could update this module to AnsibleAWSModule if you wanted. That would simplify correcting the exception handling and eliminate the need to use/import all of:
`connect_to_aws`, `ec2_argument_spec`, `get_aws_connection_info`, `boto3_conn`, and `HAS_BOTO3`

Doing this would end up looking something like:
```
from ansible.module_utils.aws.core import AnsibleAWSModule
...
module = AnsibleAWSModule(argument_spec=argument_spec)
connection = module.client('ec2')
```

And your exception handling would look like:
```
try:
    connection.describe_volumes(Filters=ansible_dict_to_boto3_filter_list(sanitized_filters))
except (BotoCoreError, ClientError) as e:
    module.fail_json_aws(e, msg="Unable to describe volumes with filters {0}".format(sanitized_filters))
```

Otherwise you end up having a mess handling the different cases of what attributes exist for what exception because they don't inherit the same properties (such as, you should include e.message and e.response if they exist, but e.message does not exist on Python 3 and BotoCoreErrors never have a response attribute).

Reply:
I didn't manage to get value for region in this case and it has to be presented in module output.

Reply:
Okay, no problem either way. You could still get region via get_aws_connection_info, which is how AnsibleAWSModule gets it too.